.github/workflows/salt-test.yml

```diff
@@ -0,0 +1,75 @@
+name: salt-test
+
+# Only run tests for main branch or if the PR has relevant changes
+on: 
+  push:
+    branches:
+      - main
+  pull_request:
+    paths:
+      - '.github/workflows/salt-test.yml'
+      - 'deployments/salt/**'
+      - 'internal/buildscripts/packaging/tests/**'
+
+concurrency:
+  group: salt-test-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+env:
+  PYTHON_VERSION: 3.8.5
+  PIP_VERSION: 20.2.4
+  REQUIREMENTS_PATH: "internal/buildscripts/packaging/tests/requirements.txt"
+  RESULT_PATH: "~/testresults"
+
+jobs:
+  salt-test:
+    name: salt-test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        PACKAGE_TYPE: [ "deb", "rpm" ]
+    steps:
+      - name: Check out the codebase.
+        uses: actions/checkout@v2
+
+      - name: Lint
```

Comment:
Let's move the lint step to a separate job since it doesn't need to be run twice.

Reply:
Thanks for the suggestions.
Added `salt-lint-test` job

Reply:
Sorry, I meant that the lint job should be separate from the test job, but they can both be in the same workflow.  Also, please add the lint job as a dependency to the test job.

Reply:
Thanks for the guidance!
I have added dependency on salt-test and salt-lint-test. can you please verify it